### PR TITLE
[FLOC-1162] Volume API version endpoint

### PR DESCRIPTION
**Pull request by wallrj**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1162

### CODE REVIEW

File: flocker/restapi/_infrastructure.py [hunk 1]
```diff
@@ -17,7 +17,8 @@
 from twisted.web.http import OK, INTERNAL_SERVER_ERROR
 
 from eliot import Logger, writeFailure
-from eliot.twisted import DeferredContext
+from eliot.twisted import DeferredContext, redirectLogsForTrial
+redirectLogsForTrial()
```
Comment (itamarst) [inline flocker/restapi/_infrastructure.py]: This should probably happen in `flocker/__init__.py`.

File: flocker/volume/schema/types.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+$schema: http://json-schema.org/draft-04/schema#
```
Comment (itamarst) [inline flocker/volume/schema/types.yml]: I suspect types is going to get pretty big pretty soon. So maybe only move stuff there that is going to appear in more than one endpoint, and inline one-off types?

File: flocker/volume/httpapi.py [hunk 3]
```diff
@@ -10,9 +13,17 @@
...
+    b'/v1/types.json': yaml.safe_load(SCHEMA_BASE.child(b'types.yml').getContent()),
+    b'/v1/endpoints.json': yaml.safe_load(SCHEMA_BASE.child(b'endpoints.yml').getContent()),
+    }
 
-class DatasetAPIUser(object):
+
+class DatasetAPIUserV1(object):
     """
     A user accessing the API.
     """
```
Comment (itamarst) [inline flocker/volume/httpapi.py]: Can probably drop noop, since we have a more useful API endpoint for testing.

**Comment by itamarst** [comment]
Looks good, thanks.

File: flocker/restapi/_infrastructure.py (hunk 1 above)
Comment (wallrj) [inline flocker/restapi/_infrastructure.py]: Done.

File: flocker/volume/schema/types.yml (hunk 2 above)
Comment (wallrj) [inline flocker/volume/schema/types.yml]: Done.

File: flocker/volume/httpapi.py (hunk 3 above)
Comment (wallrj) [inline flocker/volume/httpapi.py]: Done.

**Comment by wallrj** [comment]
Thanks @itamarst 

I addressed those comments and I'll merge when buildbot reports (mostly) success.

**Comment by wallrj** [comment]
I added the eliot global logging, but it's broken the ``flocker.__version__`` import in ``admin.vagrant``

```
Traceback (most recent call last):
  File "admin/build-vagrant-box", line 12, in <module>
    from admin.vagrant import main
  File "/home/buildslave/vagrant-slave/flocker-vagrant-tutorial-box/build/admin/vagrant.py", line 14, in <module>
    import flocker
  File "/home/buildslave/vagrant-slave/flocker-vagrant-tutorial-box/build/flocker/__init__.py", line 27, in <module>
    from eliot.twisted import redirectLogsForTrial
ImportError: No module named eliot.twisted
```
http://build.clusterhq.com/builders/flocker-vagrant-tutorial-box/builds/517/steps/build-base-box/logs/stdio

I suppose I'll make the ``redirectLogsForTrial`` call only if we are running from ``trial``.

@tomprince What do you think?

**Comment by wallrj** [comment]
This is getting me down :sob: 

```
2015/01/13 11:28:12 Error response from daemon: Cannot destroy container 8379ac32dd3a271f735308ca914bf45935283aa3820560381da7178bb2e4b272: Driver devicemapper failed to remove root filesystem 8379ac32dd3a271f735308ca914bf45935283aa3820560381da7178bb2e4b272: Device is Busy
Traceback (most recent call last):
  File "/srv/buildslave/flocker-admin/build/admin/build-package", line 13, in <module>
    main(top_level=TOPLEVEL, base_path=BASEPATH)
  File "/srv/buildslave/flocker-admin/build/admin/packaging.py", line 892, in main
    distribution=options['distribution'],
  File "/srv/buildslave/flocker-admin/build/admin/packaging.py", line 84, in run
    step.run()
  File "/srv/buildslave/flocker-admin/build/admin/packaging.py", line 683, in run
    + volume_options + [self.tag] + self.command)
  File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker', 'run', '--rm', '--volume', '/srv/buildslave/flocker-admin/build/_trial_temp/admin.test.test_packaging/OmnibusPackageBuilderTests/test_functional_fedora_20/FrvIR0/temp:/output', '--volume', '/srv/buildslave/flocker-admin/build:/flocker', 'clusterhq/build-fedora-20', '/flocker']' returned non-zero exit status 1
```

http://build.clusterhq.com/builders/flocker-admin/builds/959/steps/trial/logs/stdio

**Comment by wallrj** [comment]
Finally got successful builds (apart from the usual expected linkcheck failures). So I think I'll merge this.

**Pull request merged**